syntax = "proto3";

package user;

service UserService {
  rpc CreateUser (CreateUserRequest) returns (CreateUserResponse);
  rpc ListUsers (ListUsersRequest) returns (ListUsersResponse);
  rpc Login (LoginRequest) returns (LoginResponse);
}

message CreateUserRequest {
  string first_name = 1;
  string last_name = 2;
  string email = 3;
  string password = 4;
}

message CreateUserResponse {
  string id = 1;
  string message = 2;
}

message ListUsersRequest {
  int32 page = 1;
  int32 page_size = 2;
}

message User {
  string id = 1;
  string first_name = 2;
  string last_name = 3;
  string email = 4;
  string created_at = 5;
}

message ListUsersResponse {
  repeated User users = 1;
  int32 total = 2;
}

message LoginRequest {
  string email = 1;
  string password = 2;
}

message LoginResponse {
  string access_token = 1;
  string user_id = 2;
  string message = 3;
}